systemvm/agent/noVNC/core/rfb.js

```diff
@@ -506,13 +521,46 @@ export default class RFB extends EventTargetMixin {
 
         const scancode = XtScancode[code];
 
+        if (keysym === KeyTable.XK_Shift_L || keysym === KeyTable.XK_Shift_R) {
+            this._shiftPressed = down;
+            this._shiftKey = down ? keysym : KeyTable.XK_Shift_L;
+        }
+        if (keysym === KeyTable.XK_Control_L || keysym === KeyTable.XK_Control_R) {
+            this._ctrlPressed = down;
+        }
+        if (keysym === KeyTable.XK_Alt_L || keysym === KeyTable.XK_Alt_R) {
+            this._altPressed = down;
+        }
+
         if (this._qemuExtKeyEventSupported && scancode) {
             // 0 is NoSymbol
             keysym = keysym || 0;
 
             Log.Info("Sending key (" + (down ? "down" : "up") + "): keysym " + keysym + ", scancode " + scancode);
 
             RFB.messages.QEMUExtendedKeyEvent(this._sock, keysym, down, scancode);
+        } else if (this._VMwareCursorSupported && Object.keys(this._scancodes).length > 0) {
```

Comment:
Could it be the case in which the JP keyboard is needed but uses a different encoding?

Reply:
From I have tested, the flag is always true. I will remove it to ensure the console works if different encoding is used